src/io.rs

```diff
@@ -341,6 +361,38 @@ impl Write for &'_ mut [u8] {
     }
 }
 
+/// Converts a [`std::io::Write`] into a [`manticore::io::Write`].
+///
+/// [`write_bytes()`] is implemented by simply calling [`write()`] repeatedly
+/// until every byte is written; [`manticore::io::Write`] should be implemented
+/// directly if possible.
+///
+/// This type is provided instead of implementing [`manticore::io::Write`]
+/// directly for every [`std::io::Write`] due to trait coherence issues
+/// involving the blanket impl on `&mut _`.
+///
+/// [`std::io::Write`]: https://doc.rust-lang.org/std/io/trait.Write.html
+/// [`manticore::io::Write`]: trait.Write.html
+/// [`write_bytes()`]: trait.Write.html#tymethod.write_bytes
+/// [`write()`]: https://doc.rust-lang.org/std/io/trait.Write.html#tymethod.write
+#[cfg(feature = "std")]
+pub struct StdWrite<W>(pub W);
+
+#[cfg(feature = "std")]
+impl<W: std::io::Write> Write for StdWrite<W> {
+    fn write_bytes(&mut self, mut buf: &[u8]) -> Result<(), Error> {
+        use std::io::ErrorKind;
+        loop {
+            match self.0.write(buf).map_err(|e| e.kind()) {
```

Comment:
Do we have to implement an explicit break from the loop when buf is exhausted?

Reply:
+1?

Reply:
Yup, good catch.